File: Regular/11. Container With Most Water/Solution2.c
// Complexity: O(N)
// Two pointers, credit given to 4ms sample solution
int maxArea(int* height, int heightSize) {
    int i = 0, j = heightSize - 1, tmp = 0, vol = 0;
    while (i < j) {
        if (*(height + i) < *(height + j)) {
            tmp = *(height + i) * (j - i);
            i++;
        }
        else {
            tmp = *(height + j) * (j - i);
            j--;
        }
        if (tmp > vol) vol = tmp;
    }
    return vol;
}

File: Regular/189. Rotate Array/Solution2.java
class Solution {
    public void rotate(int[] nums, int k) {
        if (nums == null || nums.length < 2) return;
        k %= nums.length;
        if (k < 1) return;
        int[] tmp = new int[k];
        for (int i = nums.length - k; i < nums.length; i++) tmp[(i + k) % nums.length] = nums[i];
        for (int i = nums.length - 1; i >= k; i--) nums[i] = nums[i - k];
        for (int i = 0; i < k; i++) nums[i] = tmp[i];
    }
}
